ethereum/p2p/src/test/java/org/hyperledger/besu/ethereum/p2p/discovery/internal/PeerDiscoveryControllerTest.java

```diff
@@ -1371,6 +1371,45 @@ public void shouldNotRespondToENRRequestForNonBondedPeer() {
         .send(eq(peers.get(0)), matchPacketOfType(PacketType.ENR_REQUEST));
   }
 
+  @Test
```

Comment:
(optional) Might be nice to add another test that checks that we don't respond to an ENR request received way before a PONG message.  Could do this by injecting a `CacheBuilder` with a customer [ticker](https://guava.dev/releases/snapshot/api/docs/com/google/common/cache/CacheBuilder.html#ticker(com.google.common.base.Ticker)) that can be advanced from the test.